### ISSUE TITLE
extended at-kwdef

### ISSUE THREAD

**Issue by tpapp** [JuliaLang/Compat.jl#638]
`Base.@kwdef` was extended considerably by https://github.com/JuliaLang/julia/pull/29316, would it make sense for Compat.jl to provide a backport?

**Comment by christopher-dG** [comment]
Bump: I really want to use this for PkgTemplates but I want to keep supporting 1.0.

**Comment by fredrikekre** [comment]
Use Parameters.jl?

**Comment by christopher-dG** [comment]
Yeah, that's certainly an option. I would of course prefer to reach for something in Base though.
